### PR TITLE
iox-1394 fix axivion warnings in filesystem

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

To fix the Axivion warnings in `filesystem` I had to do some refactoring. The `perms` enum was replaced by a `access_control` class and the constants are moved to the `perms` namespace.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Relates to #1394

### CODE REVIEW

**Review by elfenpiff — COMMENTED** [review]
Only two tiny things and we can merge it.
1. Please rename `access_control` into `access_rights`. It makes for me more sense that the  `access_rights` consisting of a set of `perms` or permissions. `access_control` is for me the act of enforcing `access_rights` and may lead to confusion with `access_control_list`s which is not a list of `access_control`s.
2. Rename the method `value()` into `as_mode_t()` so that it is clear what kind of value one gets here. Especially in a review one does not see what `value()` really returns and how the permission is encoded there.

File: iceoryx_hoofs/source/posix_wrapper/shared_memory_object.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ expected<SharedMemoryObject, SharedMemoryObjectError> SharedMemoryObjectBuilder:
                        << ", access mode = " << asStringLiteral(m_accessMode)
                        << ", open mode = " << asStringLiteral(m_openMode)
                        << ", baseAddressHint = " << logBaseAddressHint
-                       << ", permissions = " << iox::log::oct(static_cast<mode_t>(m_permissions)) << " ]";
+                       << ", permissions = " << iox::log::oct(m_permissions.value()) << " ]";
```
[the remark this answers is not in the document]
Comment (elBoberido) [reply iceoryx_hoofs/source/posix_wrapper/shared_memory_object.cpp:78]: I thought about this too but then there is the issue that `mode_t` is 4 byte on Linux and Windows and 2 bytes on FreeBSD and macOS. Not sure we want to rely on that for the underlying type. If it is only for the method, then we could take that route but we might get some `useless cast` warnings.

**Comment by FerdinandSpitzschnueffler** [comment]
@elBoberido Can you please update the filesystem entry in the hoofs readme?

File: iceoryx_hoofs/legacy/include/iceoryx_hoofs/cxx/filesystem.hpp [hunk 2]
```diff
@@ -24,8 +24,8 @@ namespace iox
 /// [[deprecated("Deprecated in 3.0, removed in 4.0, please include 'iox/filesystem.hpp' instead")]]
 namespace cxx
 {
-/// @deprecated use `iox::perms` instead of `iox::cxx::perms`
-using iox::perms;
+/// @deprecated use 'iox::access_control' instead of 'iox::cxx::perms'
```
[the remark this answers is not in the document]
Comment (elBoberido) [reply iceoryx_hoofs/legacy/include/iceoryx_hoofs/cxx/filesystem.hpp:27]: As is turned out, it was only added to the namespace the the other legacy includes follow the same pattern like this one.

I suggest to keep it as is and when we add the deprecated attribute also add the version

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

File: iceoryx_hoofs/source/posix_wrapper/shared_memory_object.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (elfenpiff) [reply iceoryx_hoofs/source/posix_wrapper/shared_memory_object.cpp:78]: We can always change it later, so fine for now

**Review by elfenpiff — APPROVED** [review]

**Comment by elBoberido** [comment]
Aligned with @dkroenke to ignore the integration tests while the issues with the ROS setup are sorted out

**Pull request merged**